### PR TITLE
Fix Invalid Mem w/ Moving Window, LB & PMLs

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR is a workaround to fix an invalid memory access when moving window is used together with timers-based load balancing.
See issue https://github.com/ECP-WarpX/WarpX/issues/3377

### CODE REVIEW

File: Source/Utils/WarpXMovingWindow.cpp [hunk 1]
```diff
@@ -428,7 +428,9 @@ WarpX::shiftMF (amrex::MultiFab& mf, const amrex::Geometry& geom,
             dstfab(i,j,k,n) = srcfab(i+shift.x,j+shift.y,k+shift.z,n);
         })
 
-        if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+        if (cost &&
+            mfi.index() < cost->size() && //Workaround to fix invalid memory accesses when mf is a PML
```
Comment (ax3l) [inline Source/Utils/WarpXMovingWindow.cpp]: attn @atmyers @WeiqunZhang : PMLs have a different domain decomposition than regular MFs. How can we handle costs for them (see: `&(*cost)[mfi.index()]` below)?

Comment (atmyers) [reply Source/Utils/WarpXMovingWindow.cpp]: One thought is - do we actually want to track costs for the PML grids? Usually, the local cost of the PML update is small compared to other costs in the simulation, and in particular to the cost of the parallel copy from the PML grids to the parent grids (maybe this is different for PSATD or maybe there are future planned PML implementations that will be expensive, though?). This function could simply check whether it's working on PML grids and skip updating the costs if so. 

Another is that currently, we never actually apply load balancing separately to the PML grids. The default behavior is to assign the PML box to whatever rank the closest parent grid box is assigned to, in order to minimize communication costs. So if you do want include the pml cost somewhere, if `WarpX::do_similar_dm_pml = 1`, perhaps it makes the most sense to give it to whatever box the pml is "following". 

If `WarpX::do_similar_dm_pml = 0`, the PML will be load balanced using the default SFC anyway, so don't use the costs anyway. If, in the future, we do want to load balance the PML grids separately, then we'd need to track a separate set of pml_costs, defined on the pml grids, and use them in the load balancing step, but that would be a task for a separate, non-bugfix PR.

In summary, I think it's probably fine just to skip updating the costs for PML grids. If you want to do a little better than that, you could skip them if `WarpX::do_similar_dm_pml = 0` and include them in the "parent" box if "WarpX::do_similar_dm_pml = 1", but I expect that the improvement in the cost accuracy won't be significant, at least for FDTD.

Comment (ax3l) [reply Source/Utils/WarpXMovingWindow.cpp]: > This function could simply check whether it's working on PML grids and skip updating the costs if so.
> ... In summary, I think it's probably fine just to skip updating the costs for PML grids.

I think that would be a good solution for now. @lucafedeli88 can you please update accordingly once you are back from Japan? :)

Comment (ax3l) [reply Source/Utils/WarpXMovingWindow.cpp]: ping @lucafedeli88 :)

Comment (lucafedeli88) [reply Source/Utils/WarpXMovingWindow.cpp]: Sure! Sorry... I'll try to do that now...

Comment (lucafedeli88) [reply Source/Utils/WarpXMovingWindow.cpp]: @ax3l , would this be OK as a temporary workaround ?

Comment (ax3l) [reply Source/Utils/WarpXMovingWindow.cpp]: Thank you, that LGTM - @atmyers do you finalize the review and merge? :)

**Review by atmyers — APPROVED** [review]
